core/kazoo_documents/test/kz_device_test.erl

```diff
@@ -179,6 +200,38 @@ outbound_dynamic_flags_test_() ->
      }
     ].
 
+device_param_setting_test_() ->
+    {'ok', Device} = kz_device:fetch(?MASTER_ACCOUNT, ?DEVICE_1_ID),
+
+    Device1 = kz_device:set_sip_username(Device, <<"rick_and_morty">>),
```

Comment:
It seems like you could create a list of `[{Value, Setter, Getter}]` and then use a LC to create the tests. Something like:
```erlang
Setup = [{<<"rick_and_morty">>, fun kz_device:set_sip_username/2, fun kz_device:sip_username/1}
         ,...
        ],
[?_assertEqual(Value, Get(Set(Device, Value))) || {Value, Set, Get} <- Setup].
```